### PR TITLE
Set version of manually configured resources to v1alpha2

### LINKED ISSUE #122: Set version of manually configured resources to v1alpha2
### What problem are you facing?

In the full set, it's hard to know which CRDs are manually configured and have the less risk of major changes.

### How could Crossplane help solve your problem?

By marking them as `v1alpha2`, we can differentiate which APIs more mature.

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

Fixes #122 

This PR set versions of resources that have manual configuration to `v1alpha2`. Some resources (such as `azurerm_api_management`, `azurerm_data_share` etc.) have only Kind name configuration. For these resources the version was not set to `v1alpha2`. They are still in `v1alpha1`.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

The CRDs were generated with their new api versions.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: config/common/common.go [hunk 1]
```diff
@@ -110,8 +119,8 @@ func GetFullyQualifiedIDFn(serviceProvider string, keyPairs ...string) tjconfig.
 // rules first will be possible. Since there is no fixed index for key-value pairs in maps, it is not possible to place
 // rules from specific to general. Therefore, array is used here.
 var referenceRules = [][]string{
-	{"resource_group$", "/azure/v1alpha1.ResourceGroup"},
-	{"subnet$", "/network/v1alpha1.Subnet"},
+	{"resource_group$", ResourceGroupReferencePath},
```
Comment (ulucinar) [inline config/common/common.go:122]: Please be careful here if you update the existing constant as it looks like we need the relative path here (no concatenation with the module name).

File: config/common/common.go [hunk 2]
```diff
@@ -41,6 +41,15 @@ const (
 	nameReferenceKind = "name"
 	// idReferenceKind represents ID reference kind
 	idReferenceKind = "id"
+
+	// VersionV1Alpha2 is used as minimum version for all manually configured resources.
+	VersionV1Alpha2 = "v1alpha2"
```
Comment (ulucinar) [inline config/common/common.go:46]: Could you please make sure that it's okay `v1alpha1` versions of existing resources? Normally, we would have to have `v1alpha1` and `v1alpha2` versions simultaneously supported for upgrades and `v1alpha1`s removed at a later stage (e.g., at the end of deprecation period). However, I don't think upgrades are a consideration atm.

Comment (ulucinar) [inline config/common/common.go:46]: We may consider moving these constants to `apis/rconfig` package where `APISPackagePath` resides, as these are API type constants.

**Review by ulucinar — APPROVED** [review]

File: config/common/common.go [hunk 3]
```diff
@@ -41,6 +41,15 @@ const (
...
+
+	// ResourceGroupReferencePath is used as import path for ResourceGroup
+	ResourceGroupReferencePath = "/azure/v1alpha2.ResourceGroup"
```
[the remark this answers is not in the document]
Comment (sergenyalcin) [reply config/common/common.go]: Done!

File: config/common/common.go (hunk 2 above)
Comment (sergenyalcin) [reply config/common/common.go:46]: Moved to rconfig package!

**Comment by ulucinar** [comment]
Thank you @sergenyalcin.

**Pull request merged**